### PR TITLE
CEDS-2037 skip border transport and payment

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: test/unit/controllers/declaration/BorderTransportControllerSpec.scala [hunk 1]
```diff
@@ -101,69 +100,70 @@ class BorderTransportControllerSpec extends ControllerSpec {
...
-      }
+    "we are on occasional declaration journey" should {
+      def declarationFactory() = aDeclaration(withType(DeclarationType.OCCASIONAL))
+      behave like borderTransportControllerForInvalidJourney(declarationFactory)
     }
 
+    "we are on clearance declaration journey" should {
+      def declarationFactory() = aDeclaration(withType(DeclarationType.CLEARANCE))
+      behave like borderTransportControllerForInvalidJourney(declarationFactory)
+    }
```
Comment (PatrykRudnicki) [inline test/unit/controllers/declaration/BorderTransportControllerSpec.scala]: Should we use our helper `onJourney(...)`?

Comment (SteveSugden) [reply test/unit/controllers/declaration/BorderTransportControllerSpec.scala]: I just went with fixing / extending whatever style was there already.  I could have a go at refactoring I guess as we are not too rushed.

File: app/controllers/declaration/BorderTransportController.scala [hunk 2]
```diff
@@ -42,25 +42,28 @@ class BorderTransportController @Inject()(
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport with ModelCacheable {
 
-  def displayPage(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
+  private val validTypes = Seq(DeclarationType.STANDARD, DeclarationType.SUPPLEMENTARY)
+
+  def displayPage(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType(validTypes: _*)) { implicit request =>
```
Comment (ed0906) [inline app/controllers/declaration/BorderTransportController.scala]: Perhaps its worth adding an extra method to JourneyAction  that accepts an `Iterable` so you dont have to do the `:_*` everywhere

Comment (SteveSugden) [reply app/controllers/declaration/BorderTransportController.scala]: Yes, I was going to ask about doing something like that.   Consider it done.

**Comment by SteveSugden** [comment]
Tests updated if you want to take another look @bambuchaAdm @PatrykRudnicki

**Pull request merged**